template.tpl

```diff
@@ -546,7 +587,36 @@ const cleanObject = (obj) => {
 };
 
 /*
- * Returns the time property for Amplitude event
+ * A safer replacement to makeNumber.
+ * It will return a number only if:
+ *  - the input is a number
+ *  - the input is a string that can be parsed as a number
+ * In all other cases returns undefined.
+ *
+ * @param x - any argument
+ * @returns - a number or undefined
+ */
+const safeMakeNumber = (x) => {
+  const n = makeNumber(x);
+  if (n != x) {
+    // non-empty non-number string
+    // non-empty array
+    // object
+    // undefined
+    // null
+    return undefined;
+  }
+
+  if (getType(x) === 'array' || x === '') {
+    // the cases where we get false zeros
+    return undefined;
+  }
+
+  return n;
+};
```

Comment:
Still fails on the `Infinity` and `"Infinity"` cases (produces output will `null` value), but it's super edge case so I don't think it matters.

Reply:
This is true. At the moment the custom value for metric is derived from an event property, so i assume that Infinity cannot be as a value there. If in the future we allow providing a value from a variable, we can revisit to consider this case as you said!